### PR TITLE
`comupte` - fix `health_probe_id` update and network tests

### PR DESCRIPTION
**Pull request by myc2h6o**
Code:
- Add `d.HasChange("health_probe_id")` so that the update could be detected

Tests:
- `*_otherHealthProbeUpdate`: Fix by the code change
- `*_networkAcceleratedNetworking`: Adding deallocate operation in test so that the property can be updated
- `*_networkPublicIPTags`: Tune the parameter so that property could be set
- `*_updateHealthProbe`: Add a new test step to delete the VMSS then update the sku of Load Balancer, otherwise it cannot be updated.

### CODE REVIEW

File: internal/services/compute/linux_virtual_machine_scale_set_network_resource_test.go [hunk 1]
```diff
@@ -448,21 +450,6 @@ func TestAccLinuxVirtualMachineScaleSet_networkPublicIPTags(t *testing.T) {
 	})
 }
 
-func TestAccLinuxVirtualMachineScaleSet_RoutingPreferenceIPTags(t *testing.T) {
```
Comment (myc2h6o) [inline internal/services/compute/linux_virtual_machine_scale_set_network_resource_test.go:451]: This is dup with `TestAccLinuxVirtualMachineScaleSet_network`

Comment (myc2h6o) [inline internal/services/compute/linux_virtual_machine_scale_set_network_resource_test.go:451]: This is dup with `TestAccLinuxVirtualMachineScaleSet_networkPublicIPTags`

**Comment by myc2h6o** [comment]
Test result with parameter `(TestAccLinuxVirtualMachineScaleSet_network|TestAccWindowsVirtualMachineScaleSet_network|TestAccLinuxVirtualMachineScaleSet_updateHealthProbe|TestAccWindowsVirtualMachineScaleSet_updateHealthProbe|TestAccLinuxVirtualMachineScaleSet_otherHealthProbeUpdate|TestAccWindowsVirtualMachineScaleSet_otherHealthProbeUpdate)  `
image

File: internal/services/compute/windows_virtual_machine_scale_set_other_resource_test.go [hunk 2]
```diff
@@ -537,12 +537,19 @@ func TestAccWindowsVirtualMachineScaleSet_updateHealthProbe(t *testing.T) {
 				check.That(data.ResourceName).ExistsInAzure(r),
 			),
 		},
+		data.ImportStep("admin_password"),
+		// Have to remove VMSS to change the sku of load balancer
```
Comment (katbyte) [inline internal/services/compute/windows_virtual_machine_scale_set_other_resource_test.go:541]: We should try and figure out how to not require this, ie we shouldn't require the user to apply multiple times to do this

File: internal/services/compute/windows_virtual_machine_scale_set_network_resource_test.go [hunk 3]
```diff
@@ -33,6 +37,8 @@ func TestAccWindowsVirtualMachineScaleSet_networkAcceleratedNetworkingUpdated(t
 			Config: r.networkAcceleratedNetworking(data, false),
 			Check: acceptance.ComposeTestCheckFunc(
 				check.That(data.ResourceName).ExistsInAzure(r),
+				// Accelerated Networking can only be changed when VM is not running
+				data.CheckWithClientForResource(r.deallocateVirtualMachineScaleSet(), "azurerm_windows_virtual_machine_scale_set.test"),
```
Comment (katbyte) [inline internal/services/compute/windows_virtual_machine_scale_set_network_resource_test.go:41]: this is only fixing the test not the reason for its failure, users will still see the issue? so we should detect this change and then do this in the update function?

**Review by katbyte — CHANGES_REQUESTED** [review]

File: internal/services/compute/windows_virtual_machine_scale_set_network_resource_test.go (hunk 3 above)
Comment (myc2h6o) [reply internal/services/compute/windows_virtual_machine_scale_set_network_resource_test.go]: Make sense. I took another look at the document, and found the tests are using `Standard_F4` that doesn't support accelerated networking, thus the VM has to be deallocated. After changing it to the allowed size `Standard_D2s_v3` it doesn't require the VMto be stopped. I've also added the same missing update test to orchestrated vmss
Ref:
[Standard_F2](https://learn.microsoft.com/azure/virtual-machines/sizes-previous-gen#f-series)[Standard_D2s_v3](https://learn.microsoft.com/azure/virtual-machines/dv3-dsv3-series#dsv3-series)

File: internal/services/compute/windows_virtual_machine_scale_set_other_resource_test.go (hunk 2 above)
Comment (myc2h6o) [reply internal/services/compute/windows_virtual_machine_scale_set_other_resource_test.go]: I've updated the test to create two sets of Load Balancer and other dependencies, thus removing the step to remove VMSS.
As the `sku` in Load Balancer is `ForceNew`, we are always moving into a new one when updating it in the VMSS, so it shall be fine to replicate a new one for the update scenario.

**Comment by myc2h6o** [comment]
Updated test result:
image

**Comment by myc2h6o** [comment]
Hi @katbyte thanks for reviewing the pr! I've updated the code to resolve your comments, please take a look.

**Review by katbyte — APPROVED** [review]
LGTM ⚙️

**Pull request merged**